### PR TITLE
Clean up chunk coordinators and queue coordinators

### PR DESCRIPTION
**Pull request by traksag**
## Overview

**Fixes the unresolved issue in https://github.com/IntellectualSites/PlotSquared/issues/3197**

## Description

The problems are: 
1. Chunk coordinators are retained by their Bukkit tasks, because the tasks don't get cancelled when the chunk coordinator is done or gets cancelled.
2. Queue coordinators are retained by `activeQueues` in the global block queue.

The changes are as follows.

Cancel the repeating task associated to a chunk coordinator when the chunk coordinator is done or gets cancelled itself. Also clean up chunks of a chunk coordinator when it gets cancelled (if `unloadAfter` is enabled).

I didn't want to implement `isCancelled()` for chunk coordinators and I found it a bit confusing that chunk coordinators extend `PlotSquaredTask`, so I got rid of that super class. To me, `PlotSquaredTask` seem more like some abstraction of the platform task API. 

The `isCancelled()` check in progress subscribers isn't necessary, because the chunk coordinator calls  `notifyEnd()` anyway when it gets cancelled or is done.

I removed the `activeQueues` in the global block queue, because it doesn't have any use currently. This also means we don't need additional code to remove queue coordinators from the `activeQueues` when they get cancelled or are done.

Let me know what you think. I'm not sure what code gets called on non-main threads, so thread safety may require a more thoroughly review. Though I did check `processChunk()` in the Bukkit chunk coordinator and I believe it always gets called from the main thread.

With these changes applied and after clearing/setting components on big plots a couple of times, I don't see massive memory usage in VisualVM anymore for PlotSquared. Clearing and setting components seems to work fine too, so I believe these changes don't break anything.

## Checklist

- [x] I included all information required in the sections above
- [x] I tested my changes and approved their functionality
- [x] I ensured my changes do not break other parts of the code
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/PlotSquared/blob/v6/CONTRIBUTING.md)

### CODE REVIEW

**Comment by NotMyFault** [comment]
I'm not too sure whether removing public methods is a great idea, because that would be an API breaking change. You need to keep the old methods and introduce new ones and handle the delegate and/or deprecate the existing ones and schedule them for removal if they're superseded.

**Review by dordsor21 — CHANGES_REQUESTED** [review]

**Comment by dordsor21** [comment]
> I'm not too sure whether removing public methods is a great idea, because that would be an API breaking change. You need to keep the old methods and introduce new ones and handle the delegate and/or deprecate the existing ones and schedule them for removal if they're superseded.

These aren't really considered API classes, and as v6 is still technically in beta, I think it's acceptable to remove the classes, especially if they wouldn't make much sense given the changes here.

**Comment by NotMyFault** [comment]
> > I'm not too sure whether removing public methods is a great idea, because that would be an API breaking change. You need to keep the old methods and introduce new ones and handle the delegate and/or deprecate the existing ones and schedule them for removal if they're superseded.
> 
> These aren't really considered API classes, and as v6 is still technically in beta, I think it's acceptable to remove the classes, especially if they wouldn't make much sense given the changes here.

True, that may apply while being in beta still.

File: Bukkit/src/main/java/com/plotsquared/bukkit/queue/BukkitChunkCoordinator.java [hunk 1]
```diff
@@ -58,7 +60,7 @@
  * has fully consumed the chunk
  * </p>
  **/
-public final class BukkitChunkCoordinator extends ChunkCoordinator {
+public final class BukkitChunkCoordinator extends ChunkCoordinator implements Runnable {
```
[the remark this answers is not in the document]
Comment (traksag) [reply Bukkit/src/main/java/com/plotsquared/bukkit/queue/BukkitChunkCoordinator.java]: Done

**Review by NotMyFault — APPROVED** [review]

**Review by dordsor21 — APPROVED** [review]

**Pull request merged**